c10/core/DispatchKeySet.h

```diff
@@ -839,8 +837,25 @@ inline DispatchKeySet getAutocastRelatedKeySetFromBackend(BackendComponent t) {
     case BackendComponent::XPUBit:
       return autocast_xpu_ks;
     case BackendComponent::CUDABit:
-    case BackendComponent::XLABit:
       return autocast_cuda_ks;
+    case BackendComponent::XLABit:
+      return DispatchKeySet(DispatchKey::AutocastXLA);
+    case BackendComponent::IPUBit:
+      return DispatchKeySet(DispatchKey::AutocastIPU);
+    case BackendComponent::LazyBit:
+      return DispatchKeySet(DispatchKey::AutocastLazy);
+    case BackendComponent::MetaBit:
+      return DispatchKeySet(DispatchKey::AutocastMeta);
+    case BackendComponent::MPSBit:
+      return DispatchKeySet(DispatchKey::AutocastMPS);
+    case BackendComponent::HPUBit:
+      return DispatchKeySet(DispatchKey::AutocastHPU);
+    case BackendComponent::PrivateUse1Bit:
+      return DispatchKeySet(DispatchKey::AutocastPrivateUse1);
+    case BackendComponent::PrivateUse2Bit:
+      return DispatchKeySet(DispatchKey::AutocastPrivateUse2);
+    case BackendComponent::PrivateUse3Bit:
+      return DispatchKeySet(DispatchKey::AutocastPrivateUse3);
```

Comment:
Is this not trivial now?

Reply:
yep, it technically is - I vaguely remember when I first did the BackendComponent refactor, I was measuring perf and this function was a hot-path, so I pre-computed (and constexpr'd) the dispatch key set return arguments, so they don't need to be computed on the fly. I figured I'd keep that in here just in case to avoid a regression.

Reply:
But now the compute on the fly is literally just mashing together the functionality key and the backend bit, it should be faster than the jump table here

Reply:
The `DispatchKeySet()` ctr that mashes the two bits together does some extra bit twiddling so idk if it's a clear win, but that's definitely fair - it's probably not worth making the code uglier like this. I'll swap it out.

Reply:
oh well actually, I think I can just kill this function entirely as per Elias's [comment](https://github.com/pytorch/pytorch/pull/85043#pullrequestreview-1109679090). I'll do that and confirm nothing breaks.